worker/jobTypes/githubJob.js

```diff
@@ -0,0 +1,134 @@
+    let basePath = `https://github.com`;
+    if (currentJob.payload.private) {
+      basePath = `https://${process.env.GITHUB_BOT_USERNAME}:${process.env.GITHUB_BOT_PASSWORD}@github.com`;
+    }
+    return basePath;
+  }
+
+  // cleanup before pulling repo
+  async cleanup(logger) {
+    const currentJob = this.currentJob;
+    logger.save(`${'(rm)'.padEnd(15)}Cleaning up repository`);
+    try {
+      workerUtils.removeDirectory(`repos/${workerUtils.getRepoDirName(currentJob)}`);
+    } catch (errResult) {
+      logger.save(`${'(CLEANUP)'.padEnd(15)}failed cleaning repo directory`);
+      throw errResult;
+    }
+    return new Promise(function(resolve, reject) {
+      logger.save(`${'(rm)'.padEnd(15)}Finished cleaning repo`);
+      resolve(true);
+    });
+  }
+
+  async cloneRepo(logger) {
+    const currentJob = this.currentJob;
+    logger.save(`${'(GIT)'.padEnd(15)}Cloning repository`);
+    logger.save(`${'(GIT)'.padEnd(15)}running fetch`);
+    try {
+      if (!currentJob.payload.branchName) {
+        logger.save(`${'(CLONE)'.padEnd(15)}failed due to insufficient definition`);
+        throw new Error('branch name not indicated');
+      }
+      const basePath = this.getBasePath();
+      const repoPath = basePath + '/' + currentJob.payload.repoOwner + '/' + currentJob.payload.repoName;
+      await simpleGit('repos')
+        .silent(false)
+        .clone(repoPath, `${workerUtils.getRepoDirName(currentJob)}`)
+        .catch(err => {
+          console.error('failed: ', err);
+          throw err;
+        });
+    } catch (errResult) {
+      if (
+        errResult.hasOwnProperty('code') ||
+        errResult.hasOwnProperty('signal') ||
+        errResult.hasOwnProperty('killed')
+      ) {
+        logger.save(`${'(GIT)'.padEnd(15)}failed with code: ${errResult.code}`);
+        logger.save(`${'(GIT)'.padEnd(15)}stdErr: ${errResult.stderr}`);
+        throw errResult;
+      }
+    }
+    return new Promise(function(resolve, reject) {
+      logger.save(`${'(GIT)'.padEnd(15)}Finished git clone`);
+      resolve(true);
+    });
+  }
+
+  async buildRepo(logger) {
+    const currentJob = this.currentJob;
+
+    // setup for building
+    await this.cleanup(logger);
+    await this.cloneRepo(logger);
+
+    logger.save(`${'(BUILD)'.padEnd(15)}Running Build`);
+    logger.save(`${'(BUILD)'.padEnd(15)}running worker.sh`);
+
+    try {
+      const exec = workerUtils.getExecPromise();
+      const basePath = this.getBasePath();
+      const repoPath = basePath + '/' + currentJob.payload.repoOwner + '/' + currentJob.payload.repoName;
+
+      const command = `
+        cd repos/${workerUtils.getRepoDirName(currentJob)}; 
+        git checkout ${currentJob.payload.branchName}; 
+        git pull origin ${currentJob.payload.branchName};
+      `;
+
+      await exec(command);
+   
+      const commandbuild = `
+        . /venv/bin/activate && 
+        cd repos/${workerUtils.getRepoDirName(currentJob)} &&
```

Comment:
We talked about changing this to only check what type of build and then run the targets directly.